### PR TITLE
Defaults ws max_size on server to 16MB

### LINKED ISSUE #306: Add websocket max size
Uvicorn does not appear to support setting a web socket max message size. 

Without such an option an attacker can overwhelm the memory of an application running within Uvicorn (a denial of service). 

Other frameworks such as aiohttp, tornado and Sanic have max message size options. Hypercorn also has a max message size:

https://gitlab.com/pgjones/hypercorn/blob/master/hypercorn/config.py#L55

It is discussed/documented here: https://pgjones.gitlab.io/hypercorn/dos_mitigations.html#large-websocket-message

Cheers!

### PR DESCRIPTION
**Pull request by euri10**
replaces #538 as I had an issue with merges....

### CODE REVIEW

**Review by deepankarm — APPROVED** [review]

**Comment by phuong** [comment]
Thank you guys. Pls merge this

**Comment by JoanFM** [comment]
Hello @euri10 ,

Will you be able to solve the conflict?

**Comment by euri10** [comment]
updated to current master, should be an easy review :) I'd prefer to merge this over https://github.com/encode/uvicorn/pull/1027 since it's adding more tests

**Comment by euri10** [comment]
updated to current master, should be an easy review :)

File: tests/protocols/test_websocket.py [hunk 1]
```diff
@@ -457,3 +458,53 @@ async def get_subprotocol(url):
     async with run_server(config):
         accepted_subprotocol = await get_subprotocol("ws://127.0.0.1:8000")
         assert accepted_subprotocol == subprotocol
+
+
+MAX_WS_BYTES_PLUS1 = 1024 * 1024 * 16 + 1
+MAX_WS_BYTES = 1024 * 1024 * 16
```
Comment (Kludex) [inline tests/protocols/test_websocket.py]: If you want, if not it's fine:
```suggestion
MAX_WS_BYTES = 1024 * 1024 * 16
MAX_WS_BYTES_PLUS1 = MAX_WS_BYTES + 1
```

**Review by Kludex — APPROVED** [review]

**Pull request merged**